src/main/java/org/benetech/servicenet/service/impl/GeocodingResultServiceImpl.java

```diff
@@ -71,12 +71,31 @@ public void delete(UUID id) {
     }
 
     @Override
-    public List<GeocodingResult> createOrUpdateGeocodingResult(Location location, MatchingContext context) {
+    public Optional<GeocodingResult> getGeocodeForLocationIfUnique(Location location, MatchingContext context) {
```

Comment:
Why is this optional when we can still have multiple geolocations for one location?

Reply:
Because we can't store the coordinates in location if there were multiple responses from the Google API. In fact, Maybe we should store them at all, as it might confuse the partners that there is a data they didn't provide.